Newsweek homepage clone.

dipbazz opened the pull request:
This project is about building a [news week](https://www.newsweek.com/) home page clone.

The features of this project are listed as below:
1. The website is completely responsive with different breakpoint provided by bootstrap:
    - Below 575px for mobile screen
    - In between 575px and 768px larger mobile screen.
    - In between 768px and 992px for medium screen (tablet).
    - In between 992px and 1200px for laptop and desktop.
    - Above 1200px for large screen.

2. The site contains information:
    - Header with site navigation links.
    - Top Stroy
    - Featured stories.
    - The debate.
    - Opinions.
    - More stories.
    - Different articles.
    - In the magazine.
    - Editor's Pick
    - Subscribe and newsletter.
    - Footer with other useful links.

dipbazz committed:
Updated index.html with layout.

- Moved some layouts to first get the high level view of design.

Commit:
Add some elements index file

Add in the magazine, editor's pick, promotion and footer

dipbazz committed:
Add some elements and refactored header section.

- Rearranged navlinks on bottom header.
- used classes d-flex and change the direction of flex elements for header links.

Commit:
Add header style

Make logo icons and some buttons responsive

dipbazz committed:
Styled top and bottom navbar.

Used bootstrap class to style navbar:
- text-nowrap, font-weight-bold, border utility.

Custom style to style search icon and input.

Commit:
Add style

Add debate section style

dipbazz committed:
Add layout for my turn, culture and travel, and more stories.

- used bootstrap card layout to create a layout for all elements.
- modified slightly for category to match up the culture and travel section.

Commit:
Add styles to in the magazine and opinion

Add some styles using bootstrap classes

dipbazz committed:
Added Article one and article two content in html.

- Used bootstrap card layout to create artciles
- Added a tag inside image of cards
- created signup section inside article group 2
- Reduced 2 articles from article group 1 to match the content.

Commit:
Add styles to in the magazine and editor's pick

Add styles to two seactions adding bootstrap classes and custom css

dipbazz committed:
Add elements on html for article group 3 and subscribe and newslater

- used card layout on left and unordered list with flush on article group 3
- used two col for subscribe and newsletter
- made article 3 and newsletter responsive on all breakpoint

Commit:
Add footer and solve linter's problems

- Add newsweek logo file, made footer responsive
- Fixed the linter's issues on html and css

dipbazz committed:
update readme file and add app screenshot file of project

javitocor commented on README.md:
```diff
@@ -1,79 +1,87 @@
-- GitHub: [@githubhandle](https://github.com/githubhandle)
-- Twitter: [@twitterhandle](https://twitter.com/twitterhandle)
-- LinkedIn: [LinkedIn](https://linkedin.com/linkedinhandle)
+    - GitHub: [@yocosaka](https://github.com/yocosaka)
+    - Twitter: [@yocosaka](https://twitter.com/yocosaka)
+    - LinkedIn: [Yoko Saka](https://www.linkedin.com/in/yokosaka)
```
Kindly, make this contact look like the other one, it is more professional

javitocor reviewed (changes requested):
# STATUS CHANGES REQUIRED ♻️ 

Hello team!
Great job with that project! I am impressed! 👍 
You really master bootstrap! 🥇 
And only used less than 300 lines of CSS 👏 
Nice set up of the linters! 🟢 
Professional readme file! 💯 

However, we need to make some small changes before approving this project 💪 
Kindly, check my comments below 👇  apply the changes and resubmit your project for a new code review ⏩ 

## Mandatory Changes
-  The navbar must be fixed/sticky for all screen sizes, kindly make it happen
- Screen sizes smaller than 1024px:
   - The US, business, sports...sections are missing, please add it to this screen size
   - Kindly make the signup section, equal to the original, it shouldn't take 100% width
original:
![Capture](https://user-images.githubusercontent.com/56809963/95608418-5b9a3080-0a5d-11eb-991a-8e23dc801e46.PNG)
your project:
![Capture1](https://user-images.githubusercontent.com/56809963/95608440-62c13e80-0a5d-11eb-9e7f-769ae66e1ba2.PNG)
   - Kindly fix the first image in the editors pick section, must be all items equal
![Capture2](https://user-images.githubusercontent.com/56809963/95608554-8f755600-0a5d-11eb-82dd-bca2550c747c.PNG)
   - Fix the image in the newsletter section
original
![Capture3](https://user-images.githubusercontent.com/56809963/95608622-b16ed880-0a5d-11eb-8b4d-628a61f98851.PNG)
your project:
![Capture4](https://user-images.githubusercontent.com/56809963/95608664-c0558b00-0a5d-11eb-97e9-70030e100cb1.PNG)
- Screens below 768px
   - Kindly fix the footer, the social icons must be under the letters
original
![Capture5](https://user-images.githubusercontent.com/56809963/95608768-ebd87580-0a5d-11eb-95e5-74df161943ad.PNG)
your project
![Capture6](https://user-images.githubusercontent.com/56809963/95608804-fb57be80-0a5d-11eb-9dd5-c89902a1e7cc.PNG)
   - The US, business, sports...sections are missing, please add it to this screen size

Commit:
Fix responsiveness below 1024px

fixed article-3, signup-card, and editor's pick sections

dipbazz committed:
Fix readme and some responsiveness.

- instead of code block on author section created a unordered list.
- footer icons plced to left on small screen and right on large.
- fixed backgroun image position on newsletter section.

dipbazz commented:
Thank you for your code review @javitocor. We really appreciated and almost fixed everything you have mentioned for a change but we couldn't achieve the navbar sticky by using only CSS. So should we use javascript to achieve that sticky behavior or can we skip for now?

arslanbisharat reviewed (changes requested):
### Request changes 😳
This is an amazing project. You guys worked really amazing on this project but there is still some room for improvement. 
_It's okay. We are here to learn. Don't get discouraged. Work even harder.Let's make good mistakes next time. You are almost there.😄_
**Please have a look at the suggestions to get more understanding.**
### General

- [x] All **alt** attributes should have a short and meaningful image description.
- [x] Please use less CSS and more bootstrap for this project. You can easily achieve some tasks like display, hight, width flex, etc from bootstrap. Please work on it and change CSS to bootstrap. 
![image](https://user-images.githubusercontent.com/58504655/95741375-95f70e00-0ca7-11eb-94fe-1fd60591f7ba.png)

- [x] Your license link is broken in your Readme. Please add a license file to your project. 
 
### Large Screen


- [x] The top navbar should be sticky. It decreases its width and removes the date and weather on scrolling. But on your page navbar is not sticky. You can learn about sticky navbar in bootstrap [here](https://www.w3schools.com/howto/howto_js_navbar_sticky.asp) and here is a [video ](https://www.youtube.com/watch?v=qnV1wbcKymM). 

**Original** 

![Screenshot (20)](https://user-images.githubusercontent.com/58504655/95730943-596fe600-0c98-11eb-9dd1-b2a239afc5f9.png)

**Yours Clone**

![Screenshot (19)](https://user-images.githubusercontent.com/58504655/95730968-612f8a80-0c98-11eb-827d-1de36290249b.png)



- [x] **This is optional but highly recommended too**. On the original page there a sticky part.  This part makes the webpage more interesting and shows good knowledge and usage of bootstrap. But on your webpage, this page is missing. If you can Please work on it. 
![chrome-capture (34)](https://user-images.githubusercontent.com/58504655/95730503-c636b080-0c97-11eb-8d1a-434e69e14982.jpg)

- [x] On the original page, the newsletter page has the same height as the subscribe section. But on your page, that is a problem with it. Please work on it. 

**Yours Clone**

![chrome-capture (36)](https://user-images.githubusercontent.com/58504655/95731583-2417c800-0c99-11eb-8ba2-80a3af251d24.jpg)

**Original Page**

![chrome-capture (35)](https://user-images.githubusercontent.com/58504655/95731593-2712b880-0c99-11eb-9973-b82bb5fce370.jpg)

- [x] On your clone Newsweek Logo has more width as compared to the original page. And the social icons are not correctly aligned. 

### Medium(Tablet) Screen

- [x] The top navbar should be sticky on medium screen too. Below there are the images attached. Please have a look here. 

**Yours Clone**

![chrome-capture (41)](https://user-images.githubusercontent.com/58504655/95732876-c5ebe480-0c9a-11eb-9481-15ac84ac2c21.jpg)

**Original** 

![chrome-capture (40)](https://user-images.githubusercontent.com/58504655/95732884-c84e3e80-0c9a-11eb-8cfa-7f7e21186f5c.jpg)

- [x] When scrolling down, a white space appears which is not present on the original page. My recommendation is to make it as present on the original page. 

![chrome-capture (42)](https://user-images.githubusercontent.com/58504655/95733335-6d691700-0c9b-11eb-8036-8a08eab57c57.jpg)

- [x] On the original page the image has lesser height than the other images present on the left side. Please have a look below. You can remove this using bootstrap. Don't use CSS. 

**Clone**

![chrome-capture (44)](https://user-images.githubusercontent.com/58504655/95733687-ea948c00-0c9b-11eb-8fc3-205b12680469.jpg)

**Original** 

![chrome-capture (43)](https://user-images.githubusercontent.com/58504655/95733704-f1230380-0c9b-11eb-9be3-1e239c605fa7.jpg)

### Small Screen

- [x] Top navbar is also sticky for small screens. Please work on it. You can achieve this goal by bootstrap purely. 

- [x] on small screens, the whole footer is aligned vertically at the same position but in your clone, there is a margin around the links.  Can you please fix it? 
![chrome-capture (45)](https://user-images.githubusercontent.com/58504655/95734118-845c3900-0c9c-11eb-8c79-4aa2c3726d2c.jpg)

- [x] On small screens this section should be in the center. 
![chrome-capture (46)](https://user-images.githubusercontent.com/58504655/95734252-ad7cc980-0c9c-11eb-9ea9-6664a897595e.jpg)


**Please make the changes and request re-review for the given project**
Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

Commit:
Fix some styles and add script for sticky navigation

- Turn some code from custom css to bootstrap
- Add script file for sticly navigation

dipbazz committed:
Fix spacing and responsive based on code review.

- Made navbar fixed on top after scroll.
- footer logo and social icons aligment fixed.
- fixed the height of subscribe and newsletter section on large screen.
- provided some spacing to first image of in the magzine.

Commit:
Add sticky function to more stories

Add script and css to fix position of more stories section

dipbazz committed:
Add more stories sticky while scrolling.

- used js to make scrolling effect.

jubaan reviewed (changes requested):
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @yocosaka and @dipbazz! Great work!

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] MIT License link is broken.

- [x] All ```alt=""``` atributtes should have a short and meaningful image description.

- [x] Sections missing in tablet view: US, World, Business, Tech & science...
![image](https://user-images.githubusercontent.com/25774518/95767176-fd966480-0c79-11eb-9e38-39476d06d732.png)

- [x] Element doesn't match original
![image](https://user-images.githubusercontent.com/25774518/95767231-10a93480-0c7a-11eb-8ae8-a6235b920ca7.png)

- [x] Elements are not aligned to logos bottom
![image](https://user-images.githubusercontent.com/25774518/95767322-32a2b700-0c7a-11eb-88c1-4d5e23f93459.png)

- [x] Same sections missing on mobile version

- [x] Subscribe section doesn't match original
![image](https://user-images.githubusercontent.com/25774518/95767505-75fd2580-0c7a-11eb-9990-f922a44463da.png)

- [x] Newsletter section doesn't match original
![image](https://user-images.githubusercontent.com/25774518/95767563-890ff580-0c7a-11eb-9ef0-6da471d627b9.png)

- [x] Footer doesn't match original
![image](https://user-images.githubusercontent.com/25774518/95767626-a2b13d00-0c7a-11eb-9211-eb26a6577171.png)

### [OPTIONAL]  :yellow_circle:  

- [x] Prefer HTML classes over ids. This helps reusability, and also keeps the code DRY.

- [x] Update the title of the website it is not a test anymore. :wink: 

- [x] Element doesn't match original in tablet view
![image](https://user-images.githubusercontent.com/25774518/95767050-c45df480-0c79-11eb-92b1-b0523ecc5e7e.png)

### [Comments] :green_circle: 

- A great tool to test project and observe changes in the original website is the 
Browsers media query inspector. Please use it to verify the project looks as intended in different screen sizes.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

Commit:
Changed Live demo link on Readme file.

dipbazz committed:
Add license file, fix alt message to image and center fix scroll.

- fixed the issues with center fixed container on scroll.
- Added a license file to fix the broken link for license file.
- Gave some descriptive message to alt on image.

Commit:
Fix footer elements, subscribe, and newsletter

- aligned footer elements to logos bottom
- adjust position of subscribe images

Commit:
Fix the linter error on css

arslanbisharat reviewed (changes requested):
## changes Requested
You have done a great job with this project but Please have a look below. 

- [x] In your readme you said to clone this project just clone it. Please remove that part. This will help you to present your project more professionally. 

- [x] Text is overflowing in small screens. Please work on it an submit a new request for review. 
![95767505-75fd2580-0c7a-11eb-9990-f922a44463da](https://user-images.githubusercontent.com/58504655/95840781-07919380-0d5e-11eb-9d7f-71620f66a41b.png)
Please work on these two changes and submit a new request for review. 
**Regards,
Arslan Bisharat
slack: arslanbisharat**

dipbazz committed:
update readme file with new live demo link

arslanbisharat reviewed (changes requested):
### Changes Requested
Thank You for making some changes requested by me in the previous review. 
As requested by previous TSEs your website has text overflowing. Please work on it. In case you don't know how to fix that and need a 1:1 meeting with Full-TIme TSEs Please search on zendesk. 
**Regards,
Arslan Bisharat
Slack: arslanbisharat**

dipbazz commented:
Thank you every TSEs for your reviews. We have done every change that was requested. We have fixed the text overflow on the subscribe section @arslanbisharat, we don't see the problem on our browser it's completely working and exactly the same as the original. We are confused about your concern about text overflow. If you could explain it clearly then it would help us to clear our doubts.

Commit:
Fix position of subscribe text

umairarshadbutt reviewed (approved):
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
### The [clone](https://yocosaka.github.io/newsweek-clone/) shows a good understanding and usage of Bootstrap. :100:
##
- The cloned page is rendering as it should and meets all the project specifications.
- The project has a descriptive README.md and is professionally done. Good job with that.
- Pull Request has a meaningful title and a short description.
- CSS is clean. No linter errors at this point.
- The repo has followed all the guidelines there are to follow (GitHub flow etc.)
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @yocosaka and @dipbazz! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know on slack at any time. I will be happy to help.**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
